source/docs/v3/tenancy-bootstrappers.blade.md

```diff
@@ -29,31 +29,77 @@ php artisan cache:clear --tag=tenant_123
-    
 ```php
-use Stancl\Tenancy\Controllers\TenantAssetsController;
-use Stancl\Tenancy\Middleware\InitializeTenancyByDomainOrSubdomain;
+// Tenancy config (tenancy.filesystem.root_override)
+// %storage_path% gets replaced by storage_path()'s output
+// E.g. Storage::disk('local')->path('') will return "/$path_to_your_application/storage/$suffix/app"
+'root_override' => [
+    'local' => '%storage_path%/app/',
+    'public' => '%storage_path%/app/public/',
+],
+```
 
-    // other methods in the TenancyServiceProvider
+To make the tenant-aware `Storage` facade work with a custom disk, add the disk's name to `config('tenancy.filesystem.disks')` and override its root in `config('tenancy.filesystem.root_override')` as shown above.
 
-    public function boot()
-    {
-        // update the middleware used by the asset controller
-        TenantAssetsController::$tenancyMiddleware = InitializeTenancyByDomainOrSubdomain::class;
-    }
+### Assets
+
+The bootstrapper modifies the links retrieved by the `asset()` helper, so they link to the files *of the currently initialized and identified tenant*.
+
+Before using the `asset()` helper, make sure to [assign the identification middleware you're using in your app to TenantAssetsController's `$tenancyMiddleware`]({{ $page->link('configuration#static-properties') }}):
+
+```php
+// TenancyServiceProvider (don't forget to import the classes)
+
+public function boot()
+{
+    // Update the middleware used by the asset controller
+    TenantAssetsController::$tenancyMiddleware = InitializeTenancyByDomainOrSubdomain::class;
+}
 ```
 
-This bootstrapper is the most complex one, by far. We will have a — better written — explanation in v3 docs soon, but for now, refer to the 2.x docs for information about filesystem tenancy. [https://tenancyforlaravel.com/docs/v2/filesystem-tenancy/](https://tenancyforlaravel.com/docs/v2/filesystem-tenancy/)
+> Note: You can disable tenancy of `asset()` in the config (`tenancy.filesystem.asset_helper_tenancy`) and explicitly use `tenant_asset()` instead. But keep in mind that `tenant_asset()` does not respect the asset URL you set in `config('app.asset_url')` – the helper will always return your website's url suffixed by `tenancy/assets/name-of-your-asset.foo` (e.g. `https://your-site.test/tenancy/assets/asset-name.txt`).
+>
+> You may want to do that if you're facing issues using a package that utilizes `asset()` inside the tenant app.
+>
+> For non-tenant-specific assets (assets shared among all tenants or JS/CSS assets), you can use `global_asset()` and `mix()`.
```

Comment:
Perhaps mention that this is an alternative to disabling asset helper tenancy?

So change it to something like:
```suggestion
> As an alternative to disabling `asset_helper_tenancy`, you may use `global_asset()` and `mix()` for non-tenant-specific assets (assets shared among all tenants or JS/CSS assets).
```

Reply:
I think saying that it's an alternative makes it look like we're suggesting that disabling the asset helper tenancy is the primary way to go about accessing the non-tenant-specific assets. Maybe the above note about disabling asset_helper_tenancy would be enough?

Reply:
True, you can ignore this suggestion if you do the restructure I mentioned.